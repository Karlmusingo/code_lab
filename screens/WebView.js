import React, { Component } from "react";
import { WebView } from "react-native-webview";

class ProfilePage extends Component {
  state = {};
  render() {
    const { navigation } = this.props;
    const login = navigation.getParam("login", "NO-ID");
    return <WebView source={{ uri: `https://github.com/${login}` }} />;
  }
}

export default ProfilePage;
