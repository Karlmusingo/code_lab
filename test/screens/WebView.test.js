import React from "react";
import { shallow, mount } from "enzyme";
import { MockedProvider } from "react-apollo/test-utils";
import renderer from "react-test-renderer";
import WebView from "../../screens/WebView";

describe("Render the profile component", () => {
  const props = {
    navigation: {
      getParam: jest.fn(() => "login"),
      navigate: jest.fn()
    }
  };
  it("should render the profile component", async () => {
    const component = shallow(<WebView {...props} />);
    expect(component).toMatchSnapshot();
  });
});
