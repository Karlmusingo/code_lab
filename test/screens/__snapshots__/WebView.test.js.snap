// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render the profile component should render the profile component 1`] = `
<WebView
  cacheEnabled={true}
  originWhitelist={
    Array [
      "http://*",
      "https://*",
    ]
  }
  source={
    Object {
      "uri": "https://github.com/login",
    }
  }
  useSharedProcessPool={true}
  useWebKit={true}
/>
`;
